cpu/atmega2560/include/periph_cpu.h

```diff
@@ -20,12 +20,33 @@
 #define PERIPH_CPU_H_
 
 #include "periph/dev_enums.h"
+#include <avr/io.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-/* nothing defined here so far... */
+/**
+ * @brief   Define a CPU specific GPIO pin generator macro
+ */
+#define GPIO(x, y)          ((x << 4) | y)
+
+/**
+ * @brief   Available ports on the ATmega2560 family
+ */
+enum {
+    PORT_A = 0,       /**< port A */
+    PORT_B = 1,       /**< port B */
+    PORT_C = 2,       /**< port C */
+    PORT_D = 3,       /**< port D */
+    PORT_E = 4,       /**< port E */
+    PORT_F = 5,       /**< port F */
+    PORT_G = 6,       /**< port G */
+    PORT_H = 0,       /**< port H */
```

Comment:
I neither looked at the code nor checked details, but just as an assumption: Won't it be enough to just support up to PORT_F? IIRC there was a shift in the registers behind PORT_F or PORT_G and these pins aren’t available on the arduino-mega2560 anyway. (We're on the CPU specific and not board specific part here, I know I know :-) )